### PR TITLE
Add ability to uglify generated code

### PR DESCRIPTION
**Pull request by SimenB**
@mcollina Something like this you had in mind?

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 const fastSafeStringify = require('fast-safe-stringify')
 
+var uglify
```
Comment (SimenB) [inline index.js]: Seems inconsistent between `let` and `var` in the code base. Which do you want?

File: index.js [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 const fastSafeStringify = require('fast-safe-stringify')
 
+var uglify = null
```
Comment (SimenB) [inline index.js]: Seems inconsistent between `let` and `var` in the code base. Which do you want?

Comment (delvedor) [inline index.js]: `var` is better since is a little faster than `let`, thanks! :)

File: bench.js [hunk 3]
```diff
@@ -35,9 +35,9 @@ const obj = {
 
 const multiArray = []
 
-const stringify = require('.')(schema)
-const stringifyArray = require('.')(arraySchema)
-const stringifyString = require('.')({ type: 'string' })
+const stringify = require('.')(schema, { uglify: true })
+const stringifyArray = require('.')(arraySchema, { uglify: true })
+const stringifyString = require('.')({ type: 'string', uglify: true })
```
Comment (mcollina) [inline bench.js]: Can you include benchmarks for both uglified and non-uglified versions?

**Comment by mcollina** [comment]
This is also conflicting for the other two PRs, can you rebase as well?

**Comment by mcollina** [comment]
@delvedor I think we can bump v1.0.0 when this lands, it has been pretty stable API-wise for the last releases.

**Comment by delvedor** [comment]
@mcollina I agree, but before I think we should close/improve https://github.com/fastify/fast-json-stringify/pull/22, then we are ready!

**Comment by SimenB** [comment]
Rebased and separate benchmark added plus updated readme.

File: README.md [hunk 4]
```diff
@@ -7,14 +7,18 @@ advantages reduces on large payloads.
...
-JSON.stringify array x 3,679 ops/sec ±1.01% (85 runs sampled)
-fast-json-stringify array x 4,618 ops/sec ±1.64% (87 runs sampled)
-JSON.stringify long string x 13,303 ops/sec ±1.01% (89 runs sampled)
-fast-json-stringify long string x 13,489 ops/sec ±0.88% (90 runs sampled)
-JSON.stringify short string x 4,974,749 ops/sec ±1.14% (86 runs sampled)
-fast-json-stringify short string x 11,030,700 ops/sec ±0.82% (89 runs sampled)
-JSON.stringify obj x 1,774,593 ops/sec ±1.07% (90 runs sampled)
-fast-json-stringify obj x 4,976,369 ops/sec ±1.00% (89 runs sampled)
+JSON.stringify array x 3,288 ops/sec ±5.18% (82 runs sampled)
+fast-json-stringify array x 1,813 ops/sec ±10.21% (71 runs sampled)
```
Comment (SimenB) [inline README.md]: This is worse, but that's #15, right?

Comment (delvedor) [inline README.md]: Yeah, we must definitely fix that issue :S

File: package.json [hunk 5]
```diff
@@ -29,7 +29,8 @@
     "long": "^3.2.0",
     "pre-commit": "^1.1.3",
     "standard": "^10.0.0",
-    "tap": "^10.3.0"
+    "tap": "^10.3.0",
+    "uglify-es": "^3.0.9"
```
Comment (SimenB) [inline package.json]: We need the es variant to support `const` in the code, btw

**Comment by mcollina** [comment]
@SimenB would you like to help maintaining this library and join our team?

**Comment by SimenB** [comment]
@mcollina I'd love to join up! 🙂

**Comment by mcollina** [comment]
I've sent you an invitation!

**Pull request merged**